webserver/include/webserver/servercontext.h

```diff
@@ -5,32 +5,57 @@
 
 #include <boost/program_options.hpp>
 
-namespace po = boost::program_options; 
-
 namespace cc
-{  
+{
 namespace webserver
 {
 
-template <class Base>
-class PluginHandler;
+class SessionManager;
 
+/**
+ * ServerContext is given to each service request handler's constructor. This
+ * class can be used by the handlers to obtain information about the running
+ * server's configuration.
+ */
 struct ServerContext
 {
-  ServerContext(
-    const std::string& compassRoot_,
-    const po::variables_map& options_) :
-      compassRoot(compassRoot_),
-      options(options_)
+  ServerContext(const std::string& compassRoot_,
+                const boost::program_options::variables_map& options_)
+    : ServerContext(compassRoot_, options_, nullptr)
+  {
+  }
+
+  ServerContext(const std::string& compassRoot_,
+                const boost::program_options::variables_map& options_,
+                SessionManager* sessionManager_)
+    : compassRoot(compassRoot_), options(options_),
+      sessionManager(sessionManager_)
   {
   }
 
+  /**
+   * The directory where the CodeCompass server was installed to.
+   */
   const std::string& compassRoot;
-  const po::variables_map& options;
+  /**
+   * The command-line invocation options of the server, after parsing.
+   */
+  const boost::program_options::variables_map& options;
+  /**
+   * The session manager object the server is using to handle users and
+   * authentications.
+   *
+   * This is an opaque pointer to a type internal to the Web server
+   * implementation. Clients may use SessionManagerAccess from session.h to
+   * access user data.
+   */
+  // This is done to allow passing the pointer to handlers but decouple the
```

Comment:
Why did you mixed comment types here? Above you used `/**/` and here you are using `//`.

Reply:
`/**` is the documentation comment that is to be picked up by doxygen and IDEs. The `//` is a simple comment for the developers who want to understand why this (passing the opaque pointer) is done like so - it is not a trivial construct unless you know the nitty-gritty of C(++) magics.